File: backend/src/main/java/com/shoptourr/api/v1/dto/home/HomeDtos.java
package com.shoptourr.api.v1.dto.home;

import com.shoptourr.api.v1.dto.common.CommonDtos.MoneyDto;
import com.shoptourr.api.v1.dto.trip.TripDtos.TripSummaryDto;
import com.shoptourr.api.v1.dto.user.UserDtos.UserDto;

import java.util.List;

/**
 * Home screen aggregate — one round-trip for first paint.
 * Prefer this over N separate calls on cold start.
 */
public final class HomeDtos {

    private HomeDtos() {}

    public record HomeResponse(
            UserDto user,
            TripSummaryDto currentTrip,
            List<TripSummaryDto> upcoming,
            List<TripSummaryDto> archive,
            /** Sum of spent across active + past (in preferredCurrency). */
            MoneyDto allTimeSpent,
            int unreadAlertCount
    ) {}
}

File: shared/src/commonMain/kotlin/com/example/shoptourr/api/home/HomeDtos.kt
package com.example.shoptourr.api.home

import com.example.shoptourr.api.common.MoneyDto
import com.example.shoptourr.api.trip.TripSummaryDto
import com.example.shoptourr.api.user.UserDto
import kotlinx.serialization.Serializable

@Serializable
data class HomeResponse(
    val user: UserDto,
    val currentTrip: TripSummaryDto? = null,
    val upcoming: List<TripSummaryDto> = emptyList(),
    val archive: List<TripSummaryDto> = emptyList(),
    val allTimeSpent: MoneyDto,
    val unreadAlertCount: Int = 0,
)
